src/advanced_inv.cpp

```diff
@@ -1727,6 +1773,7 @@ void advanced_inventory::remove_item( advanced_inv_listitem &sitem )
         assert( &cont->contents.front() == sitem.it );
         cont->contents.erase( cont->contents.begin() );
     } else if(sitem.area == AIM_VEHICLE || sitem.area == AIM_DRAGGED) {
+        set_vehicle(panes[src], panes[src].veh_area);
         if(s.can_store_in_vehicle()) {
```

Comment:
This test seems strange: `remove_item` has no return value, nothing to inform the caller whether removing was done or not. The caller will simply assume it has been done, and it's the callers responsibility to assure removing the item is actually possible (otherwise it must not call `remove_item`).

Now this test seems to indicate there is a situation where the item removal is not possible. This will essentially lead to item duplication as the caller will add the item to the target location and afterwards remove the original item (but the removal might fail here).